typedocs: `npm run export-typedocs` throws compilation errors

**Describe the bug**
Running `npm run export-typedocs` throws following error: 
```sh
npm run export-typedocs

> export-typedocs
> npx tsc export-typedocs.ts && node export-typedocs

node_modules/typedoc/dist/index.d.ts:16:8 - error TS1259: Module '"/Users/shubham.prajapat/repos/sourcefuse/loopback4-microservice-catalog/node_modules/typescript/lib/typescript"' can only be default-imported using the 'esModuleInterop' flag

16 import TypeScript from "typescript";
          ~~~~~~~~~~

  node_modules/typescript/lib/typescript.d.ts:7273:1
    7273 export = ts;
         ~~~~~~~~~~~~
    This module is declared with 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.

node_modules/typedoc/dist/lib/converter/comments/index.d.ts:1:8 - error TS1259: Module '"/Users/shubham.prajapat/repos/sourcefuse/loopback4-microservice-catalog/node_modules/typescript/lib/typescript"' can only be default-imported using the 'esModuleInterop' flag

1 import ts from "typescript";
         ~~

  node_modules/typescript/lib/typescript.d.ts:7273:1
    7273 export = ts;
         ~~~~~~~~~~~~
    This module is declared with 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.

node_modules/typedoc/dist/lib/converter/context.d.ts:1:8 - error TS1259: Module '"/Users/shubham.prajapat/repos/sourcefuse/loopback4-microservice-catalog/node_modules/typescript/lib/typescript"' can only be default-imported using the 'esModuleInterop' flag

1 import ts from "typescript";
         ~~

  node_modules/typescript/lib/typescript.d.ts:7273:1
    7273 export = ts;
         ~~~~~~~~~~~~
    This module is declared with 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.

node_modules/typedoc/dist/lib/converter/convert-expression.d.ts:1:8 - error TS1259: Module '"/Users/shubham.prajapat/repos/sourcefuse/loopback4-microservice-catalog/node_modules/typescript/lib/typescript"' can only be default-imported using the 'esModuleInterop' flag

1 import ts from "typescript";
         ~~

  node_modules/typescript/lib/typescript.d.ts:7273:1
    7273 export = ts;
         ~~~~~~~~~~~~
    This module is declared with 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.

node_modules/typedoc/dist/lib/converter/converter.d.ts:1:8 - error TS1259: Module '"/Users/shubham.prajapat/repos/sourcefuse/loopback4-microservice-catalog/node_modules/typescript/lib/typescript"' can only be default-imported using the 'esModuleInterop' flag

1 import ts from "typescript";
         ~~

  node_modules/typescript/lib/typescript.d.ts:7273:1
    7273 export = ts;
         ~~~~~~~~~~~~
    This module is declared with 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.

node_modules/typedoc/dist/lib/models/reflections/ReflectionSymbolId.d.ts:1:8 - error TS1259: Module '"/Users/shubham.prajapat/repos/sourcefuse/loopback4-microservice-catalog/node_modules/typescript/lib/typescript"' can only be default-imported using the 'esModuleInterop' flag

1 import ts from "typescript";
         ~~

  node_modules/typescript/lib/typescript.d.ts:7273:1
    7273 export = ts;
         ~~~~~~~~~~~~
    This module is declared with 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.

node_modules/typedoc/dist/lib/utils/entry-point.d.ts:1:8 - error TS1259: Module '"/Users/shubham.prajapat/repos/sourcefuse/loopback4-microservice-catalog/node_modules/typescript/lib/typescript"' can only be default-imported using the 'esModuleInterop' flag

1 import ts from "typescript";
         ~~

  node_modules/typescript/lib/typescript.d.ts:7273:1
    7273 export = ts;
         ~~~~~~~~~~~~
    This module is declared with 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.

node_modules/typedoc/dist/lib/utils/loggers.d.ts:1:8 - error TS1259: Module '"/Users/shubham.prajapat/repos/sourcefuse/loopback4-microservice-catalog/node_modules/typescript/lib/typescript"' can only be default-imported using the 'esModuleInterop' flag

1 import ts from "typescript";
         ~~

  node_modules/typescript/lib/typescript.d.ts:7273:1
    7273 export = ts;
         ~~~~~~~~~~~~
    This module is declared with 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.

node_modules/typedoc/node_modules/minimatch/dist/cjs/ast.d.ts:4:5 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

4     #private;
      ~~~~~~~~


Found 9 errors in 9 files.

Errors  Files
     1  node_modules/typedoc/dist/index.d.ts:16
     1  node_modules/typedoc/dist/lib/converter/comments/index.d.ts:1
     1  node_modules/typedoc/dist/lib/converter/context.d.ts:1
     1  node_modules/typedoc/dist/lib/converter/convert-expression.d.ts:1
     1  node_modules/typedoc/dist/lib/converter/converter.d.ts:1
     1  node_modules/typedoc/dist/lib/models/reflections/ReflectionSymbolId.d.ts:1
     1  node_modules/typedoc/dist/lib/utils/entry-point.d.ts:1
     1  node_modules/typedoc/dist/lib/utils/loggers.d.ts:1
     1  node_modules/typedoc/node_modules/minimatch/dist/cjs/ast.d.ts:4
```

**To Reproduce**
Steps to reproduce the behavior:
1. Clone the repo
2. Run `npm install`
3. Run `npx lerna bootstrap`
4. Run `npm run export-typedocs`
5. See error

**Expected behavior**
It should export the API references in docs folder for all configured packages.

**Screenshots**
NA

**Additional context**
__

--- a/package.json
+++ b/package.json
@@ -16,8 +16,8 @@
     "cz-format-extension": "^1.5.0",
     "husky": "^7.0.4",
     "lerna": "^5.1.5",
-    "typedoc": "^0.24.8",
-    "typedoc-plugin-markdown": "^3.15.3"
+    "typedoc": "0.23.23",
+    "typedoc-plugin-markdown": "3.14.0"
   },
   "config": {
     "commitizen": {

